portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,53 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private ArrayList<String> comments = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<p>Hello Visitor!</p>");
+      response.setContentType("application/json;");
+      String json = convertToJson(this.comments);
+      response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = getParameter(request, "text-input", "");
+
+    // Add new comment only if the string is more than just white spaces
+    if(text.trim().length() > 0){
+      comments.add(text);
+    }
+
+    response.sendRedirect("/index.html");
   }
+
+    /**
+    * Converts a ServerStats instance into a JSON string using the Gson library.
```

Comment:
Update comment?  This doesn't seem to describe what we are doing, unless I am not following what a ServerStats object is.